oak-commons/pom.xml

```diff
@@ -58,6 +58,9 @@
               org.apache.jackrabbit.oak.commons.sort,
               org.apache.jackrabbit.oak.commons.properties
             </Export-Package>
+            <Embed-Dependency>
+              lz4-java;scope=compile|runtime
+            </Embed-Dependency>
```

Comment:
For this change, I think we should ask others in the Oak team... @reschke , @mreutegg , @anchela

Reply:
> This is a big change, and I think we need to be careful to not break anything. I think we will need more reviewers.

This is a PR to incorporate changes from review of https://github.com/apache/jackrabbit-oak/pull/587. So, its been reviewed by many reviewers already